fe/fe-core/src/main/java/com/starrocks/catalog/OlapTable.java

```diff
@@ -1104,6 +1104,15 @@ public Set<String> getPartitionNames() {
         return Sets.newHashSet(nameToPartition.keySet());
     }
 
+    public Set<String> getValidPartitionNames(int partitionTTLNumber) {
+        // ess than 0 means not set
+        if (partitionTTLNumber < 0) {
+            return getPartitionNames();
+        }
+        int startIndex = Math.max(0, nameToPartition.size() - partitionTTLNumber);
+        return nameToPartition.keySet().stream().skip(startIndex).collect(Collectors.toSet());
```

Comment:
is keySet sorted?

Reply:
keySet is sorted but from logical is wrong.